packages/spacecat-shared-gpt-client/src/clients/firefall-client.js

```diff
@@ -161,23 +163,113 @@ export default class FirefallClient {
     return jobStatusResponse;
   }
 
-  async fetch(prompt) {
+  /**
+   * Fetches data from Firefall Chat Completion API.
+   * @param prompt The text prompt to provide to Firefall
+   * @param options The options for the call:
+   *                - imageUrls: An array of URLs of the images to provide to Firefall
+   * @returns {Promise<*>}
+   */
+  async fetchChatCompletion(prompt, options = {}) {
     if (!hasText(prompt)) {
       throw new Error('Invalid prompt received');
     }
+    const hasImageUrls = options?.imageUrls && options.imageUrls.length > 0;
 
     try {
       const startTime = process.hrtime.bigint();
-      const jobSubmissionResponse = await this.#submitJob(prompt);
-      const jobStatusResponse = await this.#pollJobStatus(jobSubmissionResponse.job_id);
-      this.#logDuration('Firefall API call', startTime);
+      const userRole = {
+        role: 'user',
+        content: [
+          {
+            type: 'text',
+            text: prompt,
+          },
+        ],
+      };
+      const body = {
+        llm_metadata: {
+          model_name: this.config.FIREFALL_API_CAPABILITY_NAME,
```

Comment:
I think we should add a different config property for model name. `FIREFALL_API_MODEL_NAME`. Will the model name be validated against models we want to support? How can I request a response format of `json_object`?

Reply:
I'm adding `model` and `responseFormat` to options.
```
      const options = {
        model: 'gpt-4-turbo',
        responseFormat: 'json_object',
      };
```